test/fixtures/env-app/middleware/index.js

```diff
@@ -0,0 +1,8 @@
+exports.myMiddleware = function(name) {
+  return function(req, res, next) {
+    req._names = req._names || [];
+    req._names.push(name);
+    res.setHeader('names', req._names.join(','));
+    next();
+  };
+};
```

Comment:
AFAICT, this middleware is not used by your test, please remove `env-app/middleware/index.js` and also `env-app/middleware.json`.